internal/resolver/resolver.go

```diff
@@ -578,13 +598,47 @@ func (r *resolver) loadAsFileOrDirectory(path string) (string, bool) {
 	return "", false
 }
 
+func isTsConfigPathMatch(pattern string, path string) (string, bool) {
+	starIndex := strings.IndexRune(pattern, '*')
+	if starIndex == -1 {
+		firstPath := strings.Split(path, "/")[0]
+		return strings.Replace(path, pattern+"/", "", 1), firstPath == pattern
+	}
+
+	elements := strings.Split(pattern, "*")
+	if starIndex == 0 {
+		suffix := elements[1]
+		return strings.Replace(path, suffix, "", 1), strings.HasSuffix(path, suffix)
+	}
+	prefix := elements[0]
+	return strings.Replace(path, prefix, "", 1), strings.HasPrefix(path, prefix)
+}
+
 func (r *resolver) loadNodeModules(path string, dirInfo *dirInfo) (string, bool) {
 	for {
 		// Handle TypeScript base URLs for TypeScript code
-		if dirInfo.tsConfigJson != nil && dirInfo.tsConfigJson.absPathBaseUrl != nil {
-			basePath := r.fs.Join(*dirInfo.tsConfigJson.absPathBaseUrl, path)
-			if absolute, ok := r.loadAsFileOrDirectory(basePath); ok {
-				return absolute, true
+		if dirInfo.tsConfigJson != nil {
+
+			if dirInfo.tsConfigJson.absPathBaseUrl != nil {
+				if dirInfo.tsConfigJson.paths != nil {
+					for key, originalPaths := range dirInfo.tsConfigJson.paths {
+						for _, originalPath := range originalPaths {
```

Comment:
This doesn't follow the behavior of the TypeScript compiler which prefers exact matches over pattern matches and prefers matches with longer prefixes over matches with shorter prefixes. That's ok though, I'll fix it.